test/functions/test_base.rb

```diff
@@ -229,6 +229,25 @@ def test_normalize_space
       assert_equal( [REXML::Comment.new("COMMENT A")], m )
     end
 
+    def test_normalize_space2
+      source1 = "<a><b>breakfast boosts concentration</b><c>Coffee beans aroma</c><d>Dessert after dinner</d></a>"
+      source2 = <<-XML
+<a><b>breakfast     boosts\t\t
+ 
+concentration   </b><c>
+Coffee beans
+ aroma
+
+
+
+</c><d>        Dessert
+ \t\t    after    dinner</d></a>
+      XML
+      ret1 = REXML::XPath.each(REXML::Document.new(source1), "//text()").to_a
+      ret2 = REXML::XPath.each(REXML::Document.new(source2), "normalize-space(//text())").to_a
+      assert_equal( ret1, ret2 )
```

Comment:
We don't need to create an XML for expected data. We can just use an `Array` of `String`:

```suggestion
      source = <<-XML
<a><b>breakfast     boosts\t\t
 
concentration   </b><c>
Coffee beans
 aroma



</c><d>        Dessert
 \t\t    after    dinner</d></a>
      XML
      normalized_texts = REXML::XPath.each(REXML::Document.new(source), "normalize-space(//text())").to_a
      assert_equal([
                     "breakfast boosts concentration",
                     "Coffee beans aroma",
                     "Dessert after dinner",
                   ],
                   normalized_texts)
```

Reply:
Thank you for the suggestions. I have updated the pull request accordingly. Your guidance is greatly appreciated!